### ISSUE TITLE
For default/sink ingress paths HAproxy expecting a trailing slash

### ISSUE THREAD

**Issue by arun11299** [jcmoraisjr/haproxy-ingress#906]
HAProxy ingress version : 0.13.6

Kubernetes : GKE
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5-gke.1805", GitCommit:"6be6bf42dd3e740d8515b47dee3c9fc3ddf602b2", GitTreeState:"clean", BuildDate:"2022-02-18T05:04:07Z", GoVersion:"go1.16.7b7", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
```
haproxy configuration. configmap:
```
Name:         haproxy-configmap
Namespace:    ar-lp
Labels:       app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name:  ar-lp
              meta.helm.sh/release-namespace: default

Data
====
bind-https:
----
:8443
config-backend:
----
timeout server 500s

config-frontend:
----
http-response replace-header Set-Cookie (.*) \1;\ Secure

bind-http:
----
:8080
```

Ingress config:
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/backends: '{"k8s-be-32289--cb238fb942e4da41":"HEALTHY","k8s1-cb238fb9-ar-lp-ar-lp-grafana-8300-2ab57ed8":"UNHEALTHY","k8s1-cb238fb9-ar-lp-ar-lp-launchpad-8008-c5e04f6b":"UNHEALTHY","k8s1-cb238fb9-ar-lp-ar-lp-launchpad-8009-e5256e99":"UNHEALTHY","k8s1-cb238fb9-ar-lp-ar-lp-launchpad-8443-63aa6161":"UNHEALTHY"}'
    ingress.kubernetes.io/forwarding-rule: k8s2-fr-595g39yp-ar-lp-rw-ingress-eqa0flxs
    ingress.kubernetes.io/target-proxy: k8s2-tp-595g39yp-ar-lp-rw-ingress-eqa0flxs
    ingress.kubernetes.io/url-map: k8s2-um-595g39yp-ar-lp-rw-ingress-eqa0flxs
    meta.helm.sh/release-name: ar-lp
    meta.helm.sh/release-namespace: default
  creationTimestamp: "xxxxxxx"
  finalizers:
  - networking.gke.io/ingress-finalizer-V2
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: rw-ingress
  namespace: ar-lp
  resourceVersion: "132890740"
  uid: c1bfe36a-df7b-4692-9fc3-4aa79c7e3d13
spec:
  rules:
  - http:
      paths:
      - backend:
          service:
            name: ar-lp-launchpad
            port:
              number: 8443
        path: /
        pathType: Prefix
      - backend:
          service:
            name: ar-lp-launchpad
            port:
              number: 8009
        path: /token
        pathType: Prefix
```

A query to path "/login" was supposed to match with the first path, but it does not.
```
curl -k -u "admin:admin" https://xx.xx.xx.xx/login
default backend - 404
```

But "/login/" works. Also, not sure what UNHEALTHY means in the backend annotations. On what basis are they marked unhealthy ?

This used to work in the older release that we were using (v0.6).

**Comment by jcmoraisjr** [comment]
Hi thanks for reporting this behavior. I can confirm that there is some annoyance with Prefix match type (haproxy's `map_dir()`) and default host. You can circumvent this using `pathType: ImplementationSpecific` instead, which is the only one supported up to v0.11 iirc.

**Comment by jcmoraisjr** [comment]
Update: none of the annotations in the ingress resource you provided is supported and recognized by HAProxy Ingress, you can check in the doc, link [here](https://haproxy-ingress.github.io/docs/configuration/keys/#keys). Afaik the helm chart doesn't configure this for you as well.

**Comment by arun11299** [comment]
Thanks. I am using v0.11.9 now which seems to be working.

**Comment by jcmoraisjr** [comment]
Fixed in v0.12.13 and v0.13.7 which were just tagged. Thanks for the report! Closing.
